parlai/core/worlds.py

```diff
@@ -747,6 +747,34 @@ def _override_opts_in_shared(table, overrides):
     return table
 
 
+class BatchMultiwWorld(MultiWorld):
+    def __init__(self, opt, worlds):
+        World.__init__(self, opt)
+        self.worlds = worlds
+        self.world_idx = 0
+        self.new_world = True
+        self.parleys = -1
+        self.random = opt.get('datatype', None) == 'train'
+        # Make multi-task task probabilities.
+        self.cum_task_weights = [1] * len(self.worlds)
+        self.task_choices = range(len(self.worlds))
+        weights = self.opt.get('multitask_weights', [1])
+        sum = 0
+        for i in self.task_choices:
+            if len(weights) > i:
+                weight = weights[i]
+            else:
+                weight = 1
+            self.cum_task_weights[i] = weight + sum
+            sum += weight
+
+    def parley_init(self):
+        """
+        Flipping world id
```

Comment:
Can you add a more descriptive docstring here? What does "flipping world id" mean?